sycl/source/detail/kernel_bundle_impl.hpp

```diff
@@ -927,6 +930,13 @@ class kernel_bundle_impl {
     return true;
   }
 
+  ~kernel_bundle_impl() {
+    if (DeviceBinaries) {
```

Comment:
`removeImages()` might throw an exception (because `getSYCLKernelID` and several other sub-calls can)

But `~kernel_bundle_impl()` itself is implictly `noexcept`.  So you'll need a try/catch here.   You can use that `__SYCL_REPORT_EXCEPTION_TO_STREAM` macro if appropriate.

Reply:
Done, thanks!